rfcs/20210510-sum-with-zero-point/README.md

```diff
@@ -0,0 +1,97 @@
+primitive. The solution should also include the benchdnn extension.
+
+## Design
+
+Currently, the sum operation also supports scale. Zero point support should 
+be designed to keep backward compatibility.
+
+### API extension
+
+1. Overload `append_sum(scale, dt)` function by adding 
+`append_sum(scale, zp, dt)` instance for C++ API. Add 
+`dnnl_post_ops_append_sum_v3` for C API (preferred)
+    - Pros:
+        - It is backwards compatible.
+        - Allows to use zero point without scale (scale == 1.0).
+    - Cons:
+        - It is not the simplest solution
+
+2. Extend `append_sum(scale, dt)` function to 
+`append_sum(scale, zp, dt)` and `dnnl_post_ops_append_sum_v2` to support zero 
+point.
+    - Pros:
+        - Very simple solution.
+        - Universal solution.
+    - Cons:
+        - It is not backwards compatible.
+
+3. Add possibility to use `dnnl_primitive_attr_set_zero_points` with sum.
+    - Pros:
+        - It will allow to use mask and array of zero-point values (instead of 
+          one scalar value).
+    - Cons:
+        - More complicated solution.
+        - Probably requires backward incompatibility or adding a new function.
+        - Probably not needed in real topologies.
+
+### Sum zero-point design
+
+1. Use two parameters - `arg_zp` and `zero_point`:
+    - Pros:
+        - Compatible with scale in append_sum (one scalar value).
+        - Zero-point in sum is currently only needed for src (previous dst) but
+          allows to add an operation for other arguments later. It is also more
+          intuitive than making assumptions for arg.
+    - Cons:
+        - Inconsistent with `set_zero_points` function.
+        - `arg_zp` may be confused with the argument for scale.
+
+2. Use only `zero_point` parameter:
+    - Pros:
+        - Compatible with scale in append_sum (one scalar value).
+        - Avoid confusing `arg_zp` with the scale argument.
+    - Cons:
+        - Inconsistent with `set_zero_points` function.
+        - It assumes that the argument is src (previous dst), which is not
+          intuitive.
+
+3. Use parameters from `set_zero_points` function - `zp_arg`, `mask`, vector of
+   `zero_points`:
+    - Pros:
+        - Consistent with `set_zero_points` function.
+        - It allows all the possibilities of zero_point.
+        - Zero-point in sum is currently only needed for src (previous dst) but
+          allows to add an operation for other arguments later. It is also more
+          intuitive than making assumptions for arg.
+    - Cons:
+        - Incompatible with scale in append_sum.
+        - Probably more complex than necessary.
+        - `arg_zp` may be confused with the argument for scale.
+
+### Benchdnn extension
+
+1. Overwrite `SUM[:SCALE[:DATA_TYPE]]` post-ops parameter by
+`SUM[:SCALE[:ZEROPOINT[:DATA_TYPE]]]` instance. (preferred)
+    - Pros:
+        - More compatible with api function - more intuitive.
+    - Cons:
+        - Existing tests must be changed
```

Comment:
This first seem to be the more natural approach. The other way would be:
`SUM[:SCALE[:ZEROPOINT][:DATA_TYPE]]`
Casting @dzarukin for his opinion as it might complicate the parsing.

Reply:
I would vote for the first option. Thanks.

Reply:
I agree with it.